 <launch>
    <arg name="openrover_node_name" default="rr_openrover_driver" />
    <arg name="config_locks"  default="$(find rr_rover_zero_driver)/config/twist_mux_locks.yaml"/>
    <arg name="config_topics" default="$(find rr_rover_zero_driver)/config/twist_mux_topics.yaml"/>

    <!-- Rover Zero Node -->
    <node pkg="rr_rover_zero_driver" type="rover_zero.py" name="rover_zero_driver_node" output="screen">
        <param name="dev" value="/dev/rover-zero"/>
        <param name="baud" value="115200"/>
        <param name="timeout" value="0.1"/>
        <param name="attempts" value="1"/>
        <param name="max_vel" value="5.0"/>
        <param name="effort_coef" value="0.09"/>
        <param name="max_turn" value="6.28"/>
        <param name="diag_frequency" value="1.0"/>
        <param name="cmd_vel_timeout" value="0.3"/>
        <remap from="/cmd_vel" to="/cmd_vel/managed"/>
    </node>

    <!-- Xbox 360 Controller Button Mapping -->
    <node pkg="rr_control_input_manager" type="xbox_mapper.py" name="rr_xbox_mapper_node" output="screen" >
        <remap from="joy" to="/joystick" />
        <remap from="/joystick/a_button" to="/soft_estop/enable" />
        <remap from="/joystick/b_button" to="/soft_estop/reset" />
        <remap from="/joystick/y_button" to="/joy_priority" />
        <remap from="/joystick/x_button" to="/pause_navigation"/>
        <param name="max_vel_fwd" value="1.0" />
        <param name="max_vel_turn" value="6.28" />
        <param name="adjustable_throttle" value="True" />
        <param name="drive_increment" value="20" />
        <param name="enable_flippers" value="False" />
        <param name="flipper_increment" value="20" />
        <param name="x_button_toggle" value="true"/>
        <param name="y_button_toggle" value="true"/>
    </node>

    <!--  Twist Mux  -->
    <node pkg="twist_mux" type="twist_mux" name="twist_mux" output="screen">
        <remap from="cmd_vel_out" to="/cmd_vel/managed"/>

        <rosparam file="$(arg config_locks)"  command="load"/>
        <rosparam file="$(arg config_topics)" command="load"/>
    </node>

    <!-- Control input manager -->
    <node pkg="rr_control_input_manager" type="control_input_manager.py" name="rr_control_input_manager_node" output="log" >
        <param name="driver" value="xboxdrv"/>
        <param name="wired_or_wireless" value="wireless"/>
        <rosparam file="$(find rr_control_input_manager)/config/input_topics.yaml" command="load"/>
    </node>

    <!-- Xbox controller driver -->
    <node pkg="joy" type="joy_node" name="joy_node" output="screen" respawn="true" >
        <param name="autorepeat_rate" value="10" />
        <remap from="/joy" to="/joystick" />
        <param name="dev" value="/dev/input/jsX"/>
    </node>
</launch>
